Update py-pint versions

lpritch-lanl opened the pull request.

Commit:
Updated the prefered version to 0.11 and added releases 0.9-0.10.1

Commit:
Run autopep8 to fix flake8 errors

adamjstewart reviewed (changes requested):
Missing the following dependencies:
```
python_requires = >=3.6
install_requires = setuptools
setup_requires = setuptools; setuptools_scm
```
Also, is there a reason to switch from PyPI to GitHub?

Commit:
Added python version requirements

Commit:
Replaced github with the PyPI web site.

lpritch-lanl commented:
> Missing the following dependencies:
> 
> ```
> python_requires = >=3.6

I believe I have fixed this.

> install_requires = setuptools
> setup_requires = setuptools; setuptools_scm
> ```
> 

I don't know how to resolve these. I have a depends_on('py-setuptools') statement in the package.py file, but could not locate a setuptools-scm package.

> Also, is there a reason to switch from PyPI to GitHub?

I originally had issues downloading from PyPI that I resolved. Switched it back to PyPI.

adamjstewart commented:
> I don't know how to resolve these. I have a depends_on('py-setuptools') statement in the package.py file, but could not locate a setuptools-scm package.

There is a `py-setuptools-scm` package, you just need to add a build dependency on that.

adamjstewart commented on var/spack/repos/builtin/packages/py-pint/package.py:
```diff
@@ -12,9 +12,19 @@ class PyPint(PythonPackage):
+    version(
+        '0.9', sha256='32d8a9a9d63f4f81194c0014b3b742679dce81a26d45127d9810a68a561fe4e2')
+    version(
+        '0.8.1', sha256='afcf31443a478c32bbac4b00337ee9026a13d0e2ac83d30c79151462513bb0d4')
```
These can be on a single line, we exempt checksums from line length requirements.

lpritch-lanl replied:
Placed them back on the same line as version

Commit:
Moved the sha256 up to one line. Added the py-setuptools-scm dependency

adamjstewart commented on var/spack/repos/builtin/packages/py-pint/package.py:
```diff
@@ -12,9 +12,16 @@ class PyPint(PythonPackage):
 
-    depends_on('py-setuptools', type='build')
+    depends_on('python@3.6:', type=('build', 'run'), when='@0.10:')
+    depends_on('python', type=('build', 'run'), when='@0.8.1,0.9')
```
This line isn't necessary, all `PythonPackage`s depend on Python

Commit:
Removed python dependency for 0.8.1 and 0.9 versions. Not needed since all Python packages have this dependency
